Add boolean syntax helpers

JFriel opened the pull request:
Add syntax helpers for boolean values.

jas88 reviewed (approved):
@JFriel LGTM - I've added the new properties to IQueryHelper and added a unit test, plus moved the common case up to the common ancestor with just PG overriding, hope that's OK with you?

JFriel commented:
> @JFriel LGTM - I've added the new properties to IQueryHelper and added a unit test, plus moved the common case up to the common ancestor with just PG overriding, hope that's OK with you?

Thanks for taking a look, all looks good

rkm commented:
Thanks James x2 !
